torch/utils/file_baton.py

```diff
@@ -44,14 +44,15 @@ def wait(self):
         passed to the constructor.
         """
         has_warned = False
+        should_warn = isinstance(self.warn_after_seconds, (int, float))
 
         start_time = time.time()
         while os.path.exists(self.lock_file_path):
             time.sleep(self.wait_seconds)
 
-            if isinstance(self.warn_after_seconds, (int, float)):
+            if should_warn and not has_warned:
                 waited_too_long = time.time() - start_time > self.warn_after_seconds
```

Comment:
nit: don't create an extra `waited_too_long` variable since its already obvious what it is doing and its only used once.